src/test/java/tests/configurations/EnvConfigurationTests.java

```diff
@@ -0,0 +1,92 @@
+
+    private static final String LANGUAGE_KEY = "logger.language";
+    private static final String ELEMENT_CACHE = "elementCache.isEnabled";
+    private static final String NEW_BOOL_VALUE = "true";
+    private static final String NEW_STRING_VALUE = "ru";
+    private static final String CONDITION_TIMEOUT_KEY = "timeouts.timeoutCondition";
+    private static final String NEW_INT_VALUE = "10000";
+    private static final String RETRY_NUMBER_KEY = "retry.number";
+
+    @BeforeMethod
+    public void before() {
+        System.setProperty(LANGUAGE_KEY, NEW_STRING_VALUE);
+        System.setProperty(CONDITION_TIMEOUT_KEY, NEW_INT_VALUE);
+        System.setProperty(RETRY_NUMBER_KEY, NEW_INT_VALUE);
+        System.setProperty(ELEMENT_CACHE, NEW_BOOL_VALUE);
+        CustomAqualityServices.initInjector(new TestModule(CustomAqualityServices::getApplication));
+    }
+
+    @Test
+    public void testShouldBePossibleToOverrideLanguageWithEnvVariable() {
+        SupportedLanguage language = CustomAqualityServices.getServiceProvider().getInstance(ILoggerConfiguration.class).getLanguage();
+        assertEquals(language, SupportedLanguage.RU, "Current language should be overridden with env variable");
+    }
+
+    @Test
+    public void testShouldBePossibleToOverrideTimeoutWithEnvVariable() {
+        long conditionTimeout = CustomAqualityServices.getServiceProvider().getInstance(ITimeoutConfiguration.class).getCondition();
+        assertEquals(conditionTimeout, Long.parseLong(NEW_INT_VALUE), "Condition timeout should be overridden with env variable");
+    }
+
+    @Test
+    public void testShouldBePossibleToOverrideRetryConfigurationWithEnvVariable() {
+        int retryNumber = CustomAqualityServices.getServiceProvider().getInstance(IRetryConfiguration.class).getNumber();
+        assertEquals(retryNumber, Long.parseLong(NEW_INT_VALUE), "Number of retry attempts should be overridden with env variable");
+    }
+
+    @Test
+    public void testShouldBePossibleToOverrideElementCacheStateWithEnvVariable() {
+        boolean isEnabled = CustomAqualityServices.getServiceProvider().getInstance(IElementCacheConfiguration.class).isEnabled();
+        assertTrue(isEnabled, "Element cache state should be overridden with env variable");
+    }
+
+    @Test
+    public void testNumberFormatExceptionShouldBeThrownIfTimeoutIsNotANumber() {
+        System.setProperty(CONDITION_TIMEOUT_KEY, NEW_STRING_VALUE);
+        checkNumberFormatException(() -> CustomAqualityServices.getServiceProvider().getInstance(ITimeoutConfiguration.class).getCommand());
+    }
+
+    @Test
+    public void testNumberFormatExceptionShouldBeThrownIfRetryConfigurationIsNotANumber() {
+        System.setProperty(RETRY_NUMBER_KEY, NEW_STRING_VALUE);
+        checkNumberFormatException(() -> CustomAqualityServices.getServiceProvider().getInstance(IRetryConfiguration.class).getNumber());
+    }
+
+    @AfterMethod
+    public void after() {
```

Comment:
not sure that this overriding of base class's method would work as expected

Reply:
fixed